main/broadcasters.c

```diff
@@ -305,93 +305,166 @@ static broadcaster_ops_t eddystone_ops = {
+} __attribute__((packed)) mijia_data_entry_t;
 
-static mijia_temp_hum_t *mijia_temp_hum_data_get(uint8_t *adv_data,
-    uint8_t adv_data_len, uint8_t *mijia_temp_hum_len)
+typedef struct {
+    uint16_t service_uuid;
+    uint16_t frame_ctrl;
+    uint16_t device_type;
+    uint8_t message_counter;
+    mac_addr_t mac;
+} __attribute__((packed)) mijia_header_t;
+
+static mijia_header_t *mijia_sensor_data_get(uint8_t *adv_data,
+    uint8_t adv_data_len, uint8_t *mijia_sensor_len)
 {
     uint8_t len;
     uint8_t *data = esp_ble_resolve_adv_data(adv_data,
         ESP_BLE_AD_TYPE_SERVICE_DATA, &len);
 
-    if (mijia_temp_hum_len)
-        *mijia_temp_hum_len = len;
+    if (mijia_sensor_len)
+        *mijia_sensor_len = len;
 
-    return (mijia_temp_hum_t *)data;
+    return (mijia_header_t *)data;
 }
 
-static int mijia_temp_hum_is_broadcaster(uint8_t *adv_data, size_t adv_data_len)
+static int mijia_sensor_is_broadcaster(uint8_t *adv_data, size_t adv_data_len)
 {
     uint8_t len;
-    mijia_temp_hum_t *mijia_temp_hum = mijia_temp_hum_data_get(adv_data,
+    mijia_header_t *mijia_sensor = mijia_sensor_data_get(adv_data,
         adv_data_len, &len);
 
-    if (!mijia_temp_hum || len < offsetof(mijia_temp_hum_t, data) ||
-        le16toh(mijia_temp_hum->service_uuid) != MIJIA_TEMP_HUM_SERVICE_UUID)
+    if (!mijia_sensor || len < sizeof(mijia_header_t) ||
+        le16toh(mijia_sensor->service_uuid) != MIJIA_SENSOR_SERVICE_UUID)
     {
         return 0;
     }
 
     return 1;
 }
 
-static void mijia_temp_hum_metadata_get(uint8_t *adv_data, size_t adv_data_len,
+static void mijia_sensor_metadata_get(uint8_t *adv_data, size_t adv_data_len,
     int rssi, broadcaster_meta_data_cb_t cb, void *ctx)
 {
     char s[7];
     uint8_t len;
-    mijia_temp_hum_t *mijia_temp_hum = mijia_temp_hum_data_get(adv_data,
+    mijia_header_t *mijia_header = mijia_sensor_data_get(adv_data,
         adv_data_len, &len);
 
-    cb("MACAddress", _mactoa(mijia_temp_hum->mac), ctx);
-    sprintf(s, "%hhu", mijia_temp_hum->message_counter);
+    cb("MACAddress", _mactoa(mijia_header->mac), ctx);
+    sprintf(s, "%hhu", mijia_header->message_counter);
     cb("MessageCounter", s, ctx);
-    if (mijia_temp_hum->data_type == MIJIA_TEMP_HUM_DATA_TYPE_TEMP)
-    {
-        sprintf(s, "%.1f",
-            (int16_t)le16toh(*(uint16_t *)mijia_temp_hum->data) / 10.0);
-        cb("Temperature", s, ctx);
-    }
-    else if (mijia_temp_hum->data_type == MIJIA_TEMP_HUM_DATA_TYPE_HUM)
-    {
-        sprintf(s, "%.1f", le16toh(*(uint16_t *)mijia_temp_hum->data) / 10.0);
-        cb("Humidity", s, ctx);
-    }
-    else if (mijia_temp_hum->data_type == MIJIA_TEMP_HUM_DATA_TYPE_BATT)
-    {
-        sprintf(s, "%u", *mijia_temp_hum->data);
-        cb("BatteryLevel", s, ctx);
-    }
-    else if (mijia_temp_hum->data_type == MIJIA_TEMP_HUM_DATA_TYPE_TEMP_HUM)
+
+    if ((be16toh(mijia_header->frame_ctrl) & 0x4000) == 0)
+        return; /* No data */
```

Comment:
I, generally, prefer to avoid comments along side code, could you please move it above the condition as something like: `/* Check if any data is available */`